calico-vpp-agent/policy/policy_server.go

```diff
@@ -1095,6 +1119,162 @@ func (s *Server) handleHostMetadataRemove(msg *proto.HostMetadataRemove, pending
 	return nil
 }
 
+func (s *Server) handleHostMetadataV4V6Update(msg *proto.HostMetadataV4V6Update, pending bool) (err error) {
+	ip, ip4net, err := net.ParseCIDR(msg.Ipv4Addr + "/32")
+	if err != nil {
+		return err
+	}
+	ip4net.IP = ip
+	ip, ip6net, err := net.ParseCIDR(msg.Ipv6Addr + "/32")
+	if err != nil {
+		return err
+	}
+	ip6net.IP = ip
+	localNodeSpec := &common.LocalNodeSpec{
+		Name:        msg.Hostname,
+		Labels:      msg.Labels,
+		IPv4Address: ip4net,
+		IPv6Address: ip6net,
```

Comment:
Shouldn't those two be `net.IP` instead of `*net.IPNet` ?

Reply:
@sknat  this is used in `getProviderType` in connectivity server to know if it is crossSubnet or not, that's why it is an IPNet

Reply:
Right, but from 
````go
ip, ip4net, err := net.ParseCIDR(msg.Ipv4Addr + "/32")
````
we don't have a `net.IPNet` provided to us no ?

Reply:
Oh, but we drop the prefix 😬 in 
https://github.com/projectcalico/calico/pull/7042/files#diff-a2054f157c98114eb0de4d9785e04c51f0eb2dda5bf76e6ca94dab5a659c9c0bR140-R141